### PR TITLE
Exponentiation with squaring implemented in Python3

### PR DESCRIPTION
**Pull request by ankit-rawani**
Exponentiation with squaring implemented in Python3, a/c to the guidelines. #603 #

### CODE REVIEW

File: algorithms/ar-expsq/python3/exponentiation_by_squaring.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def exponentiation_by_squaring(base, power):
+    res = 1
+
+    while(power > 0):
+        if(power % 2 != 0):
+            res *= base  #if power is ODD, multiply res with base
+
+        base *= base   # square the base
+        power = power // 2  # halving power (integer)
+
```
Comment (RavinduSachintha) [inline algorithms/ar-expsq/python3/exponentiation_by_squaring.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356288)

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@ankit-rawani would you be able to add this contribution to the readme file?

**Comment by ankit-rawani** [comment]
Okay I'll add it

**Comment by ankit-rawani** [comment]
Do I need to add my name in the contributors section as well, or it'll be done automatically?

**Comment by RavinduSachintha** [comment]
> Do I need to add my name in the contributors section as well, or it'll be done automatically?

No. Just add a tick. Contribution will be added automatically.

**Comment by ankit-rawani** [comment]
Cool

**Comment by RavinduSachintha** [comment]
@all-contributors please add @ankit-rawani for code and doc

**Comment by RavinduSachintha** [comment]
@ankit-rawani thank you for your contribution. Keep it up 👍 
Make sure to add a star to this repo 😉

**Pull request merged**